app/assets/javascripts/nested_field.js.coffee

```diff
@@ -0,0 +1,39 @@
+(($) ->
+  Blacklight.onLoad ->
+    $('.multi_value.nested-field.form-group').manage_fields()
+    $(".multi_value.nested-field.form-group").on("managed_field:remove", (event, removed) ->
+      # Remove if the nested field is an existing element and is set to be destroyed or removed
+      removed = $(removed)
+      id_field = removed.find("input[name$='[id]']")
+      if(id_field.length > 0)
+        destroy_field = $("<input type='hidden'>")
+        destroy_field.attr("name", id_field.attr("name").replace(/\[id\]/,"[_destroy]"))
+        destroy_field.val(true)
+        destroy_field.insertAfter(id_field)
+        removed.hide()
+        removed.appendTo($(this))
+    )
+    $(".multi_value.nested-field.form-group").on("managed_field:add", (event) ->
+      # Find the last child field and update the index (increment its value) in the name and id attributes so that they
+      # follow these patterns needed for a nested attribute:
+      #   name = "#{object_name}[#{attribute_name}_attributes][#{index}][label]"
+      #   id = "#{object_name}_#{attribute_name}_attributes_#{index.to_s}_label"
+      # example:
+      #   name = "article_nested_related_items_attributes][1][label]"
+      #   id = "article_nested_related_items_attributes_1_label"
+      window.setTimeout(->
+        target = $(event.target)
+        last_child = target.find(".field-wrapper").last()
+        last_child.find("div[class$='_id']").remove()
```

Comment:
I feel like it would be nice to have a method to `removeLastChildId(last_child)` which finds and removes the last childs ID.